### ISSUE TITLE
Worx adapter Error "No connection with the host"

### ISSUE THREAD

**Issue by balou65** [iobroker-community-adapters/ioBroker.worx#856]
**Describe the bug**  
Since update to version 2.3.4 the adapter dont get conected with the host

**To Reproduce**  
Adapter installed, input ID and Password, select "Worx", done

**Expected behavior**  
As soon as adaptor started, error happened as shown in the screenshot below. Phone app and Worx Landroid working well

**Screenshots & Logfiles**  
![image](https://github.com/iobroker-community-adapters/ioBroker.worx/assets/44126564/25a102de-da6b-4c20-a02a-781e77b354e8)

``
2024-03-25 17:51:55.830 - info: host.IOBrokerPC iobroker host.IOBrokerPC Adapter "system.adapter.worx.0" is started
2024-03-25 17:51:55.834 - info: host.IOBrokerPC "system.adapter.worx.0" enabled
2024-03-25 17:51:55.886 - info: host.IOBrokerPC instance system.adapter.worx.0 started with pid 9496
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: node:internal/modules/cjs/loader:1338
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: return process.dlopen(module, path.toNamespacedPath(filename));
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: ^
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]:
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: Error: Das angegebene Modul wurde nicht gefunden.
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: \\?\C:\ioBroker\node_modules\aws-crt\dist\bin\win32-x64-cruntime\aws-crt-nodejs.node
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module._extensions..node (node:internal/modules/cjs/loader:1338:18)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module.load (node:internal/modules/cjs/loader:1117:32)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module._load (node:internal/modules/cjs/loader:958:12)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module.require (node:internal/modules/cjs/loader:1141:19)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at require (node:internal/modules/cjs/helpers:110:18)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Object. (C:\ioBroker\node_modules\aws-crt\dist\native\binding.js:104:19)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module._compile (node:internal/modules/cjs/loader:1254:14)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module.load (node:internal/modules/cjs/loader:1117:32)
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: at Module._load (node:internal/modules/cjs/loader:958:12) {
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: code: 'ERR_DLOPEN_FAILED'
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: }
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]:
2024-03-25 17:51:56.625 - error: host.IOBrokerPC Caught by controller[0]: Node.js v18.15.0
2024-03-25 17:51:56.625 - error: host.IOBrokerPC instance system.adapter.worx.0 terminated with code 1 (JS_CONTROLLER_STOPPED)
2024-03-25 17:51:56.625 - info: host.IOBrokerPC Restart adapter system.adapter.worx.0 because enabled
2024-03-25 17:51:56.903 - info: host.IOBrokerPC iobroker exit 0
``

**Versions:**  
 - Adapter version: <2.3.4>
 - JS-Controller version: <4.0.24> 
 - Node version: <v18.15.0> 
 - Operating system: Windows10
 - NPM: 9.5.0(10.2.4)

**Additional context** Sorry for maybe wrong description, i am beginner

**Comment by mcm1957** [comment]
Please correct js-controller version. 6.x.x does not (yet) exist. js-controlelr is NOT javascript adapter. You see js-controller evsion at hosts display

![image](https://github.com/iobroker-community-adapters/ioBroker.worx/assets/28575778/bed56d1c-39eb-4fc5-b5cc-cc6ac97d6625)

In addition please fill out Operating system version.

**Comment by balou65** [comment]
Now correct version, thanks for explanation

**Comment by balou65** [comment]
Any ideas?

**Comment by Ilovegym66** [comment]
problem is explained and solved here: https://forum.iobroker.net/post/1144855

**Comment by mcm1957** [comment]
More exact link:
https://forum.iobroker.net/topic/73471/worx-adapter-l%C3%A4uft-nicht-mehr/11
https://forum.iobroker.net/topic/73471/worx-adapter-l%C3%A4uft-nicht-mehr/12

An iob fix should solve the problem.

From Forum:

```
Die gute Nachricht ist, dass der Adapter unter windows läuft. Die schlechte ist, dass du leider ein schlecht aktualisiertes System hast. Auch Linux User haben diese Fehlermeldung node_modules\aws-crt\dist\bin... Hier reicht meist ein iob fix...Sollte das System zu alt sein muss immer erst die Node und dann der JS-Controller aktualisiert werden. Das aws-crt Package gehört zu aws-iot-device-sdk-v2 was mir von Worx vorgegeben wird. Wir hatten auch aws-iot-device-sdk versucht, aber Worx trennt dann alle 15 Minuten die Verbindung. Daher kann ich nicht mehr zurück...

Leider kenne ich mich mit Windows nicht aus aber vielleicht kann [@sigi234](https://forum.iobroker.net/uid/2601) sagen, ob es ein iob fix für Windows gibt. Wenn nein, dann bleibt dir nur das:

Worx unter Adapter deinstallieren (nicht nur unter Instanz)
Volles Backup erstellen
ioBroker deinstallieren
iobroker [neu installieren](https://forum.iobroker.net/topic/63610/test-iobroker-unter-windows-installieren-2023-edition?_=1711567792982)
Worx installieren
Backup aufspielen
Gruß//Lucky
```

The official ioBroker Windows Installer will execute a iob fix and can be used to try to repair a broken installation.
See https://forum.iobroker.net/topic/63610/test-iobroker-unter-windows-installieren-2023-edition

**Comment by Lucky-ESA** [comment]
Fixed with 3.0.0
